ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/AzureIntegrationForm.tsx

```diff
@@ -53,16 +68,17 @@ export const validationSchema = yup.object().shape({
                             // eslint-disable-next-line @typescript-eslint/ban-ts-comment
                             // @ts-ignore
                             context?.from[2]?.value?.updatePassword || false;
+                        const useWorkloadId = context?.parent?.wifEnabled;
 
-                        if (!requirePasswordField) {
+                        if (!requirePasswordField || useWorkloadId) {
                             return true;
                         }
 
                         const trimmedValue = value?.trim();
                         return !!trimmedValue;
                     }
                 ),
-            insecure: yup.bool(),
+            wifEnabled: yup.boolean(),
```

Comment:
Nit: not a huge deal, as they are functionality equivalent, but we have 54 uses of `bool()` in the codebase, with only 11 of `boolean()`. Personally I am in favor of `boolean` as it matches the type name used by JavaScript/TypeScript, but it does introduce some inconsistency.